contracts/examples/adder/interact/src/adder_interact.rs

```diff
@@ -0,0 +1,181 @@
+        let mut interactor = Interactor::new(config.gateway())
+            .await
+            .with_tracer(INTERACTOR_SCENARIO_TRACE_PATH)
+            .await;
+        let wallet_address = interactor.register_wallet(test_wallets::mike());
+
+        Self {
+            interactor,
+            wallet_address,
+            state: State::load_state(),
+        }
+    }
+
+    async fn deploy(&mut self) {
+        let mut typed_sc_deploy = self
+            .state
+            .default_adder()
+            .init(BigUint::from(0u64))
+            .into_blockchain_call()
+            .from(&self.wallet_address)
+            .code_metadata(CodeMetadata::all())
+            .contract_code("file:../output/adder.wasm", &InterpreterContext::default())
+            .gas_limit("70,000,000")
+            .expect(TxExpect::ok());
+
+        self.interactor.sc_deploy(&mut typed_sc_deploy).await;
+
+        let result = typed_sc_deploy.response().new_deployed_address();
+        if result.is_err() {
+            println!("deploy failed: {}", result.err().unwrap());
+            return;
+        }
+
+        let new_address_bech32 = bech32::encode(&result.unwrap());
+        println!("new address: {new_address_bech32}");
+
+        let new_address_expr = format!("bech32:{new_address_bech32}");
+        self.state.set_adder_address(&new_address_expr);
+    }
+
+    async fn multi_deploy(&mut self, count: &u8) {
+        if *count == 0 {
+            println!("count must be greater than 0");
+            return;
+        }
+        println!("deploying {count} contracts...");
+
+        let mut steps = Vec::new();
+        for _ in 0..*count {
+            let typed_sc_deploy = self
+                .state
+                .default_adder()
+                .init(BigUint::from(0u64))
+                .into_blockchain_call()
+                .from(&self.wallet_address)
+                .code_metadata(CodeMetadata::all())
+                .contract_code("file:../output/adder.wasm", &InterpreterContext::default())
+                .gas_limit("70,000,000")
+                .expect(TxExpect::ok());
+
+            steps.push(typed_sc_deploy);
+        }
+
+        self.interactor
+            .multi_sc_exec(StepBuffer::from_sc_deploy_vec(&mut steps))
+            .await;
+
+        for step in steps.iter() {
+            let result = step.response().new_deployed_address();
+            if result.is_err() {
+                println!("deploy failed: {}", result.err().unwrap());
+                return;
+            }
+
+            let new_address_bech32 = bech32::encode(&result.unwrap());
+            println!("new address: {new_address_bech32}");
+        }
+    }
+
+    async fn feed_contract_egld(&mut self) {
+        let _ = self
+            .interactor
+            .transfer(
+                TransferStep::new()
+                    .from(&self.wallet_address)
+                    .to(self.state.adder())
+                    .egld_value("0,050000000000000000"),
```

Comment:
Not sure, but `0,050000000000000000` value seems odd. I think it should be `50000000000000000`.

Reply:
It's a trick, commas are ignored by the mandos format. It's just hinting at how the value should be read in 18-decimal fixed point.